File: DONGUK/프로그래머스_WEEK_13/최대공약수와 최소공배수.py
def solution(n, m):
    answer = []
    
    N = min(n, m)
    
    gcd = 1
    q1, q2 = 1, 1
    for i in range(1, N+1):
        if n % i == 0 and m % i == 0:
            gcd = i
            q1 = n // i
            q2 = m // i
    
    answer.append(gcd)
    answer.append(gcd * q1 * q2)
    return answer

File: JUNHEE/프로그래머스_week13/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    int a, b;
    int temp = 999;
    int gcd;
    if(n < m){
        a = m;
        b = n;
    }
    else{
        a = n;
        b = m;
    }
    while(true){
        temp = a % b;
        if(temp == 0){
            gcd = b;
            break;
        }
        a = b;
        b = temp;
    }
    answer.push_back(gcd);
    answer.push_back(n*m/gcd);
    return answer;
}
